sig-robotics/proposal/Generating Action Sequences for Robot Task Plans using LLM/README.md

```diff
@@ -0,0 +1,167 @@
+
+2. Perform closed-loop verification of the developed algorithms in **Gazebo** simulation environment.
+
+3. Deploy the data transmission model and other development processes using RoboSDK.
+
+
+# 3. Proposals
+By building RoboAgent and its related components on RoboSDK, RoboSDK can have the ability to handle complex robot tasks based on LLM (Large Language Models).
+
+The scope of RoboAgent and its related suite includes:
+- Building a RoboAgent capable of task planning, environment observation, and phenomena reasoning based on prompt techniques such as ReAct, self-ask, tree-of-thoughts, to enable complex task planning and processing abilities.
+- Providing a generic interface for large language models (LLM) to empower RoboAgent with inference and planning capabilities.
+- Building the RoboToolKit suite based on prompt techniques like ReAct, which interfaces with the control interface built on RoboSDK to provide action-related instructions and control capabilities to RoboAgent.
+- Providing the ability for RoboAgent to perceive the environment by accessing the sensor interface data of RoboSDK.
+- Providing relevant documentation, sample templates, and auxiliary tools to reduce the learning curve for users.
+- Creating related test cases and test reports.
+
+This design targets developers aiming to build complex robots based on LLM using RoboSDK.
+
+# 4. Design Details
+
+## Architecture Design
+To better build the capabilities of RoboAgent in task planning, environment perception, and reasoning, and to enable RoboAgent to handle complex robot instructions, we need to build the RoboAgent and related tool suites, including Planning, Action, Reasoning, and Observation. The following diagram illustrates the system architecture built based on this concept.
+
+![System Architecture](./images/image_2.png)
+
+
+
+In this architecture, after RoboAgent receives a user input, it performs task planning to generate a sequence of sub-tasks. Then, RoboAgent constructs the required actions based on the sub-task sequence and executes the Robot instructions through the relevant controller. After execution, RoboAgent observes the environment to observe the changes that occurred as a result of the executed actions. For example, the distance to the car ahead is 5 meters (thing, distance, x, y, z). Based on the observed data, the Robot performs next-action reasoning to determine whether to execute the next sub-task or revise/update the task planning. This process continues until RoboAgent determines the completion of all sub-tasks, marking the entire task as finished and ending the task scheduling process.
+
+> It should be noted that the Planning, Action, and Observation capabilities of RoboAgent are driven by large language models combined with techniques like ReAct and self-ask.
```

Comment:
Please propose one or two perception and planning algorithms we are going to use.